### ISSUE TITLE
"System error 67 has occurred." with SSHDroid Android Server

### ISSUE THREAD

**Issue by evx73** [billziss-gh/sshfs-win#87]
## Bug Report

I get "System error 67 has occurred." when connecting to a server running on a mobile Android device using SSHDroid.

I can successfully connect to the server with WinSCP.

I tested this with both:

* sshfs-win-2.7.17334-x64 / winfsp-1.2.17346
* sshfs-win-3.5.19106-x64 / winfsp-1.5.19090

On the Windows 10 client side, I am using this command and I am getting the following output:

```
> net use P: \\sshfs\root@mobile\
The password is invalid for \\sshfs\root@mobile\.

Enter the username for 'sshfs': root
Enter the password for sshfs:
System error 67 has occurred.

The network name cannot be found.
```

I can see from the server log that the connection is established, the authentication is successful, but connection is then immediately closed.

### How to Reproduce

* on Android device, setup SSHDroid and start server on standard port 22 (grant root access)
* on Windows 10 x64 client, install sshfs and winfsp
* from command prompt, execute net use P: \\sshfs\root@mobile\

### Behaviors

Remote server should be mounted to drive.

### Environment

- OS version and build: Windows 10 64bits Version 1803 (OS Build 17134.706)
SSHDroid
- WinFsp version and build:
* sshfs-win-2.7.17334-x64 / winfsp-1.2.17346
* sshfs-win-3.5.19106-x64 / winfsp-1.5.19090

**Comment by billziss-gh** [comment]
> Is there anything I can do to avoid this issue from happening? Or to validate your hypothesis?

If you are familiar with gdb under Cygwin you could debug this by placing a breakpoint on `statfs` and/or `getattr`. Upon initialization of the file system you should calls on those operations for the root directory (`"/"`). These should fail with `EACCES` or similar error code (e.g. `EPERM`).

SSHFS should be compiled with the `-g` switch to include gdb debug information.

> Is the "Access Denied" on the Windows client side or on the remote SSH server side?

This is unclear because it appears to happen inside the SSHFS layer. Likely the server side. Perhaps you can look at server logs to determine what happens.

> About moving the issue to SSHFS-Win, I understand that only repository administrators can move issues across repositories.

Done. Ping @evx73 in case there are problems with transferring issues (it is currently in Beta according to the UI GitHub presents to me).

**Comment by evx73** [comment]
I get `missing host` when running sshfs.exe from a Cygwin shell irrespective of arguments. What am I missing?

```
[/cygdrive/c/Program Files/SSHFS-Win/bin]
$ ./sshfs.exe cmd root@mobile z: -p 22 -d -o uid=-1,gid=-1
missing host
see `/usr/bin/sshfs -h' for usage
```

**Comment by billziss-gh** [comment]
Try `./sshfs.exe cmd root@mobile: z: -p 22 -d -o uid=-1,gid=-1`. (You are missing a colon (`:`) after the `root@mobile` part.

**Comment by evx73** [comment]
Same problem. I get `missing host` even when I run `./sshfs.exe -h`.

```
$ ./sshfs.exe cmd root@mobile z: -p 22 -d -o uid=-1,gid=-1
missing host
see `/usr/bin/sshfs -h' for usage
```

```
$ ./sshfs.exe -h
missing host
see `/usr/bin/sshfs -h' for usage
```

**Comment by evx73** [comment]
I found the issue with `missing host` error. The Cygwin DLLs distributed with SSHFS-WIN were clashing with my Cygwin installation DLLs. I have now removed all DLLs in SSHFS-WIN apart from cygfuse-2.8.dll from the execution PATH.

If I run sshfs.exe through gdb, I get `no debugging symbols found`. Please see below. It looks like that sshfs was not compiled with the gdb debug information.

```
$ gdb /cygdrive/c/Program\ Files/SSHFS-Win/bin/sshfs.exe
GNU gdb (GDB) (Cygwin 8.1.1-1) 8.1.1
Copyright (C) 2018 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-pc-cygwin".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from /cygdrive/c/Program Files/SSHFS-Win/bin/sshfs.exe...(no debugging symbols found)...done.
(gdb)

```

**Comment by billziss-gh** [comment]
> The Cygwin DLLs distributed with SSHFS-WIN were clashing with my Cygwin installation DLLs.

Yes, this is a potential problem. This is why in the latest SSHFS-Win BETA I rewrote the [`sshfs-win.c`](https://github.com/billziss-gh/sshfs-win/blob/master/sshfs-win.c) wrapper to avoid such problems (assuming that it gets started by a Windows program (such as the Command Prompt) and not a Cygwin program (such as bash)).

> If I run sshfs.exe through gdb, I get no debugging symbols found.

Yes, `sshfs.exe` does not have debugging symbols. This is on purpose although I should probably rethink this decision. You will have to build sshfs yourself from a Cygwin prompt. Upstream sshfs now builds on Cygwin.

**Comment by evx73** [comment]
I download the zip with the release version code. It is missing the "sshfs" sources. Not sure how to get the sshfs sources together with sshfs-win. Is the build process documented somewhere?

**Comment by adizhol** [comment]
+1

**Comment by victor302** [comment]
I also was not able initially map network drive, but then I have realized that the root directory of the mapped file system is the home directory of the user connected to the remote system. So, the folder /home/user/mydir on the remote server should be mapped as \\sshfs\user@server\mydir. For example, the folder \etc can be mapped as \\sshfs\user@server\..\..\etc

**Comment by evx73** [comment]
Hey @victor302 - are you also using SSHDroid as a server? Can you share your set-up and configuration?
